runtime/interpreter/interpreter.go

```diff
@@ -832,6 +832,8 @@ func (interpreter *Interpreter) resultValue(returnValue Value, returnType sema.T
 				resultAuth(returnType),
 				returnValue.value,
 				optionalType.Type,
+				// result must be a resource to get here, so it will never be a reference
+				EmptyLocationRange,
```

Comment:
Here and below: Pass the proper location range to the function (`resultValue`) and use it